packages/access-api/src/routes/validate-email.js

```diff
@@ -151,60 +153,29 @@ async function authorize(req, env) {
     if (confirmation.error) {
       throw new Error(`unable to validate access session: ${confirmation}`)
     }
-    if (confirmation.capability.with !== env.signer.did()) {
-      throw new Error(`Not a valid access/confirm delegation`)
-    }
-
-    // Create a absentee signer for the account that authorized the delegation
-    const account = Absentee.from({ id: confirmation.capability.nb.iss })
-    const agent = Verifier.parse(confirmation.capability.nb.aud)
-
-    // It the future we should instead render a page and allow a user to select
-    // which delegations they wish to re-delegate. Right now we just re-delegate
-    // everything that was requested for all of the resources.
-    const capabilities =
-      /** @type {ucanto.UCAN.Capabilities} */
-      (
-        confirmation.capability.nb.att.map(({ can }) => ({
-          can,
-          with: /** @type {ucanto.UCAN.Resource} */ ('ucan:*'),
-        }))
-      )
 
-    // create an delegation on behalf of the account with an absent signature.
-    const delegation = await ucanto.delegate({
-      issuer: account,
-      audience: agent,
-      capabilities,
-      expiration: Infinity,
-      // We include all the delegations to the account so that the agent will
-      // have delegation chains to all the delegated resources.
-      // We should actually filter out only delegations that support delegated
-      // capabilities, but for now we just include all of them since we only
-      // implement sudo access anyway.
-      proofs: await collect(
-        env.models.delegations.find({
-          audience: account.did(),
-        })
-      ),
-    })
-
-    const attestation = await Access.session.delegate({
-      issuer: env.signer,
-      audience: agent,
-      with: env.signer.did(),
-      nb: { proof: delegation.cid },
-      expiration: Infinity,
+    const confirm = provide(
+      Access.confirm,
+      async ({ capability, invocation }) => {
+        return accessConfirm.handleAccessConfirm(
+          /** @type {Ucanto.Invocation<import('@web3-storage/access/types').AccessConfirm>} */ (
+            invocation
+          ),
+          env
+        )
+      }
+    )
+    const confirmResult = await confirm(request, {
+      id: env.signer.verifier,
+      principal: Verifier,
     })
-
-    // Store the delegations so that they can be pulled with access/claim
-    // The fact that we're storing proofs chains that we pulled from the
-    // database is not great, but it's a tradeoff we're making for now.
-    await env.models.delegations.putMany(delegation, attestation)
-
-    const authorization = delegationsToString([delegation, attestation])
-    // Send delegations to the client through a websocket
-    await env.models.validations.putSession(authorization, agent.did())
+    if (confirmResult.error) {
+      throw new Error('error confirming')
```

Comment:
I forget, does passing `cause` to this do anything? is `confirmResult.error` swallowed if we don't pass it along or does it get printed/preserved somewhere?

Reply:
> does passing cause to this do anything?

Which `cause`?

> is confirmResult.error swallowed if we don't pass it along

yes

Reply:
whoop sorry, meant to say "does passing this along as `cause` do anything?"

like:

```
throw new Error('error confirming', {cause: confirmError.result})
```

I think I've seen this used in one or another of our codebases but hadn't confirmed it actually does anything useful

Reply:
It's a new js feature. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause
And yeah I should probably use it here

Reply:
https://github.com/web3-storage/w3protocol/pull/530/commits/9080cfb57d3aae099b21d48a5597118b99258fba

Reply:
@travis one thing to keep in mind is that the version of typescript in use needs to have updated builtin `Error` types to allow passing `error.cause`. I think most of our repos are upgraded to it now, but some may not be (yet)

Reply:
ok thx!